Source/WebCore/platform/graphics/ImageDecoder.cpp

```diff
@@ -147,4 +152,21 @@ bool ImageDecoder::supportsMediaType(MediaType type)
     return false;
 }
 
+bool ImageDecoder::frameMetadataAtIndex(size_t index, SubsamplingLevel subsamplingLevel, const DecodingOptions& options, ImageFrame& frame) const
```

Comment:
The name is a bit vague. From the call site it's hard to see that this is filling some members of `frame`. Maybe `fetchFrameMetaDataAtIndex()`?